internal/pkg/platform/schema/compiler/extension/primarykey/template/mutation.tmpl.patch

```diff
@@ -0,0 +1,80 @@
+diff --git
+--- a/mutation.tmpl
++++ b/mutation.tmpl
+@@ -154,7 +154,36 @@ // SetID sets the value of the id field. Note that this
+ 		// operation is only accepted on creation of {{ $n.Name }} entities.
```

Comment:
Ako som uz vysvetloval, realne ma entity:
- `id` field
- a potom este composed primary key fields, napr. `branchID`, `componentID`, `configID`.

Preto musime upravit `mutation.tmpl`, ktora obsahuje definicie setters pre tieto fields.

Tj. ak sa zavola `SetID(BranchKey)`, tak sa musia nastavit aj jednotlive fields.
A opacne, ak sa zavola `SetComponentID`, tak sa musi upravit aj dana cast `ID`.

To bolo ukazane v testoch hned na zaciatku: [Basic model: test base operations](https://github.com/keboola/keboola-as-code/pull/1198/commits/020e372bc5460e7344d77e2b904c94d9dbac9ddc)

Reply:
Problem moze nastat, ak v `entgo` zmenia tuto sablonu.

Preto sa na diff/patch pouziva `git diff` (ktory je mozne pouzit aj ako patch nastroj, mimo git repa).

Vyhoda je, ze `git diff` je fuzzy, a teda nejaky posun kodu, alebo zmeny v inych metodach nezabrania aplikovaniu patch-u.

+ Vysledok je pokryty testami.

-----------

To mi malo minimalizovat potrebu upravovat tuto sablonu.

Reply:
![image](https://user-images.githubusercontent.com/19371734/220051503-af24f98c-4276-4b98-a44c-bf0022020fd8.png)